libbeat/idxmgmt/ilm/std.go

```diff
@@ -104,11 +104,16 @@ func (m *stdManager) CheckEnabled() (bool, error) {
 
 func (m *stdManager) EnsureAlias() error {
 	log := m.log
+	if !m.checkExists {
```

Comment:
Should this also check `!overwrite`? The docs say:

https://github.com/elastic/beats/blob/bbf7a581722ccf7247f07031be6aecb602e5f034/metricbeat/metricbeat.reference.yml#L2100-L2103

So I think it's fair to say that if both are false, then no ILM policy is created. Still weird config, but at least consistent with the docs.

I think the alternative is to tell users to set `setup.ilm.enabled:false`, and set `output.elasticsearch.index` to the write index.

Reply:
As agreed in https://github.com/elastic/beats/pull/27508#issuecomment-902511187 I will rather correct the behaviour of the options, not work around existing bad practice.

Reply:
It's fine by me either way, but for the record:

The config is nominally about ILM _setup_, but it's being overloaded to also configure the output index name. Seems to me this is the source of the problem. The user in https://github.com/elastic/beats/issues/26322 wants to be able to configure the output index to work with ILM without setting up ILM.

Any fix other than introducing new config is going to be less than ideal, so I guess this one is fine.

Reply:
You are correct. I have responded to the original issue with your suggestions. However, I think my code still makes sense.